### PR TITLE
Fix replication transaction chain ordering writing

### PR DESCRIPTION
**Pull request by ghost**
# Description

Fix the replication transaction chain (due to regression)

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/archethic/replication.ex [hunk 1]
```diff
@@ -108,17 +106,14 @@ defmodule Archethic.Replication do
...
-            end)
-          end
+          # Stream the insertion of the chain
+          tx
+          |> stream_previous_chain()
+          |> Stream.reject(&Enum.empty?/1)
+          |> Stream.each(&TransactionChain.write/1)
+          |> Stream.run()
+
+          TransactionChain.write(Enum.filter([previous_tx, tx], & &1))
```
Comment (Neylix) [inline lib/archethic/replication.ex:116]: writing previous transaction is useless here as it is done before by stream previous chain

**Review by Neylix — APPROVED** [review]

File: lib/archethic/replication.ex (hunk 1 above)
Comment (samuelmanzanera) [reply lib/archethic/replication.ex:116]: 👍

File: lib/archethic/replication.ex [hunk 2]
```diff
@@ -108,17 +106,14 @@ defmodule Archethic.Replication do
...
-            end)
-          end
+          # Stream the insertion of the chain
+          tx
+          |> stream_previous_chain()
+          |> Stream.reject(&Enum.empty?/1)
+          |> Stream.each(&TransactionChain.write/1)
+          |> Stream.run()
+
+          TransactionChain.write_transaction(tx)
```
Comment (Neylix) [inline lib/archethic/replication.ex:116]: telemetry should be calculated after DB writing no ?

**Review by Neylix — COMMENTED** [review]
There is also a test error.
in sync test a mockDB on `write_transaction_chain` is defined to send an event, but we don't always use `write_transaction_chain`. In the test case it should be `write_transaction`

File: lib/archethic/replication.ex (hunk 2 above)
Comment (samuelmanzanera) [reply lib/archethic/replication.ex:116]: in fact this telemetry it's about the replication validation.
But I can add one for the full write

**Review by Neylix — APPROVED** [review]

**Pull request merged**